pkg/stagemanager/stage_manager.go

```diff
@@ -375,7 +371,7 @@ func (stm *StageManager) runAfterStopStage() {
 // 1: failed to start
 // 4: run before-stop/graceful-stop callback failed
 func (stm *StageManager) Stop() {
-	if !stm.started {
+	if stm.state != Running {
```

Comment:
```suggestion
	if stm.state == Nil {
```
we should allow Stop in all other states.